### PR TITLE
Integration test for delete transaction by ID #181

### PR DESCRIPTION
**Pull request by Dharakkkk**

### CODE REVIEW

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    @BeforeEach
+    void clearDatabase() {
+        financialTransactionRepository.deleteAll();
+        walletRepository.deleteAll();
+    }
+
+    @DisplayName("Should return deleted transaction successfully")
+    @Test
+    public void testDeleteFinancialTransactionById_whenDeletedFinancialTransactionWithExistingId_thenReturnDeletedSuccesfully()
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java:38]: You don't need the `public` modifier for a test method.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransaction;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+public class DeleteTransactionByIdTest extends BaseIntegrationTestIT {
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java:21]: You don't need `public` modifier for the class.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    }
+
+    @DisplayName("Should return is Not Found error when Id does not exist in a database")
+    @Test
+    void testDeleteFinancialTransactionById_whenFinancialTransactionIdIsIncorrect_thenShouldReturnNotFoundError() throws Exception {
+        Wallet wallet = walletRepository.save(new Wallet("Test wallet"));
+        FinancialTransaction testFinancialTransaction = createTestFinancialTransaction(wallet, "Test Transaction");
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/transactions/{id}", 100L)
+                .accept(MediaType.APPLICATION_JSON)).andExpect(MockMvcResultMatchers.status().isNotFound()).andReturn();
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java:54]: Do we need the `andReturn()` method here?

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    @DisplayName("Should return deleted transaction successfully")
+    @Test
+    public void testDeleteFinancialTransactionById_whenDeletedFinancialTransactionWithExistingId_thenReturnDeletedSuccesfully()
+            throws Exception {
+        Wallet wallet = walletRepository.save(new Wallet("Test wallet"));
+        FinancialTransaction testFinancialTransaction = createTestFinancialTransaction(wallet, "Test Transaction");
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/transactions/{id}", testFinancialTransaction.getId())
+                        .accept(MediaType.APPLICATION_JSON)).andExpect(MockMvcResultMatchers.status().isNoContent());
+        Assertions.assertEquals(0, financialTransactionRepository.count());
+
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java:45]: Improve code formatting

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    private FinancialTransaction createTestFinancialTransaction(Wallet wallet, String description) {
+        return financialTransactionRepository.save(FinancialTransaction.builder()
+                .wallet(wallet)
+                .amount(new BigDecimal("5.0"))
+                .date(Instant.ofEpochSecond(1L))
+                .type(FinancialTransactionType.INCOME)
+                .description(description)
+                .build());
+    }
+
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteTransactionByIdTest.java:67]: Improve code formatting

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

**Comment by dawciobiel** [comment]
L:47 not needed blank line - and there should not be.
Same with L:73 - there are two blank lines, and there should be only one

**Review by mirekgab — APPROVED** [review]

**Review by dawciobiel — COMMENTED** [review]
Looks good to me

**Pull request merged**